### PR TITLE
Fetch Product Info V1

### PR DESCRIPTION
**Pull request by ethanwayda22**
# Description
Follow up to https://github.com/BoltApp/bolt-magento2/pull/1223

-Add parent/children lookup for configurable items
-Add stock to each of those products fetched
- remove hmac validation from get request

Fixes: (link Jira ticket)

#changelog WIP: fetch inventory v1

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [X] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [X] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# For PR Reviewer 
- [ ] Reviewed unit tests to make sure we are using real components rather than mocks as much as possible?
- [ ] For any major change (observer, new Bolt feature, core Magento interaction) we must add a feature switch, did you verify this?

# Checklist:

- [X] My code follows the style guidelines of this project.
- [X] I have performed a self-review of my own code.
- [X] I have commented my code, particularly in hard-to-understand areas.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have created or modified unit tests to sufficiently cover my changes.
- [ ] I have added my Jira ticket link and provided a changelog message.

### CODE REVIEW

**Review by vitaliyreznikov — APPROVED** [review]

**Comment by danac-gs** [comment]
Please check my comment here https://github.com/BoltApp/bolt-magento2/pull/1223#discussion_r613009241

File: Model/Api/GetProduct.php [hunk 1]
```diff
@@ -65,60 +91,120 @@ class GetProduct implements GetProductInterface
...
         $this->stockRegistry = $stockRegistry;
+        $this->stockConfiguration = $stockConfiguration;
         $this->productData = $productData;
         $this->storeManager = $storeManager;
         $this->hookHelper = $hookHelper;
         $this->bugsnag = $bugsnag;
+        $this->configurable = $configurable;
+    }
+
+    private function getStockStatus($product){
```
Comment (ethanwayda22) [inline Model/Api/GetProduct.php:122]: @danac-gs had to add this function for backwards compadibilty. Would love your thoughts on this.

I followed this article https://www.mexbs.com/magento-blog/getting-the-stock-status-programatically-in-magento-2-with-and-without-msi/

Comment (ethanwayda22) [reply Model/Api/GetProduct.php:122]: Merging now to test, will circle back to this comment in seperate PR

File: Model/Api/GetProduct.php [hunk 2]
```diff
@@ -65,60 +91,120 @@ class GetProduct implements GetProductInterface
...
+        $stockStatus = $this->stockRegistry->getStockStatus($product->getId(), $this->websiteId);
```
Comment (danac-gs) [inline Model/Api/GetProduct.php:123]: `\Magento\CatalogInventory\Api\StockRegistryInterface::getStockStatus` seems to provide even better output, although it is missing individual stock items for each of the stock sources.

File: Model/Api/GetProduct.php [hunk 3]
```diff
@@ -65,60 +91,120 @@ class GetProduct implements GetProductInterface
...
+            $productInventory->setProduct($product);
+            $productInventory->setStock($this->getStockStatus($product));
+            $this->productData->setProductInventory($productInventory);
+        }
+    }
+
+
+    private function getProductFamily(){
+        $product = $this->productData->getProductInventory()->getProduct();
+        $parent = $this->configurable->getParentIdsByChild($product->getId());
```
Comment (danac-gs) [inline Model/Api/GetProduct.php:150]: There are also Grouped and Bundle products that can have children

File: Model/Api/GetProduct.php [hunk 4]
```diff
@@ -65,60 +91,120 @@ class GetProduct implements GetProductInterface
...
+
+            $children = $parentProduct->getTypeInstance()->getUsedProducts($parentProduct);
+            $childrenStockArray = array();
+            foreach ($children  as $child) {
+                $childProductInventory = new ProductInventoryInfo();
+                $childProductInventory->setProduct($child);
+                $childProductInventory->setStock($this->getStockStatus($child));
+                array_push($childrenStockArray, $childProductInventory);
+            }
+            $this->productData->setChildren($childrenStockArray);
```
Comment (danac-gs) [inline Model/Api/GetProduct.php:167]: In this case, they are technically siblings

File: Model/Api/GetProduct.php [hunk 5]
```diff
@@ -65,60 +91,120 @@ class GetProduct implements GetProductInterface
...
+        } elseif ($product->getTypeId() == "configurable") {
```
Comment (danac-gs) [inline Model/Api/GetProduct.php:168]: } elseif ($product->getTypeId() == Configurable::TYPE_CODE) {

File: Model/Api/GetProduct.php [hunk 6]
```diff
@@ -65,60 +91,120 @@ class GetProduct implements GetProductInterface
...
+            $children = $product->getTypeInstance()->getUsedProducts($product);
+            $childrenStockArray = array();
+            foreach ($children  as $child) {
+                $childProductInventory = new ProductInventoryInfo();
+                $childProductInventory->setProduct($child);
+                $childProductInventory->setStock($this->getStockStatus($child));
+                array_push($childrenStockArray, $childProductInventory);
+            }
+            $this->productData->setChildren($childrenStockArray);
```
Comment (danac-gs) [inline Model/Api/GetProduct.php:177]: This seems duplicate and can by DRYed out

File: Model/Api/GetProduct.php [hunk 7]
```diff
@@ -65,60 +91,120 @@ class GetProduct implements GetProductInterface
...
         }
 
         try {
-            $storeId = $this->storeManager->getStore()->getId();
-            $product = $this->productRepositoryInterface->getById($productID, false, $storeId, false);
-            $this->productData->setProduct($product);
-            $stockItem = $this->stockRegistry->getStockItem($product->getId());
-            $this->productData->setStock($stockItem);
+            $this->getProduct($productID, $sku);
+            $this->getProductFamily();
```
Comment (danac-gs) [inline Model/Api/GetProduct.php:203]: These don't get or return anything, so an alternative naming would be better, like generate or collect

**Pull request merged**